app/src/main/java/com/keylesspalace/tusky/adapter/NotificationsAdapter.java

```diff
@@ -124,12 +124,24 @@ public NotificationsAdapter(AdapterDataSource<NotificationViewData> dataSource,
 
     @Override
     public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int position) {
+        bindViewHolder(viewHolder,position,null);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int position, @NonNull List payloads) {
+        bindViewHolder(viewHolder,position,payloads);
+    }
+
+    private void bindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int position, @Nullable List payloads){
+        Object payloadForHolder = payloads!=null&&!payloads.isEmpty()?payloads.get(0):null;
         if (position < this.dataSource.getItemCount()) {
             NotificationViewData notification = dataSource.getItemAt(position);
             if (notification instanceof NotificationViewData.Placeholder) {
-                NotificationViewData.Placeholder placeholder = ((NotificationViewData.Placeholder) notification);
-                PlaceholderViewHolder holder = (PlaceholderViewHolder) viewHolder;
-                holder.setup(statusListener, placeholder.isLoading());
+                if (payloadForHolder==null) {
```

Comment:
Why is it important to check payload for placeholder?

Reply:
@charlag If payloadForHolder==null - need to update a whole ViewHolder. If payloadsForHolder!=null and it is a List - enum all items of this list and update only required fields. It may contain only a single field for update now.
`
public static class Key{
        public static final String KEY_CREATED = "created";
}
`

But may be easy extended at the future.